shared/constants/types/chat2/message.js

```diff
@@ -63,7 +63,7 @@ export type _MessageText = {
   errorReason: ?string,
   hasBeenEdited: boolean,
   id: MessageID,
-  submitState: null | 'deleting' | 'editing' | 'pending',
```

Comment:
We were implicitly representing `failed` by having a non-null `errorReason`. This makes it explicit, and allows us to show a message in a failed state without having an associated error message.